osu.Game/Online/API/Requests/SearchBeatmapSetsRequest.cs

```diff
@@ -16,15 +16,19 @@ public class SearchBeatmapSetsRequest : APIRequest<SearchBeatmapSetsResponse>
         private readonly BeatmapSearchCategory searchCategory;
         private readonly DirectSortCriteria sortCriteria;
         private readonly SortDirection direction;
+        private readonly BeatmapSearchGenre genre;
+        private readonly BeatmapSearchLanguage language;
         private string directionString => direction == SortDirection.Descending ? @"desc" : @"asc";
 
-        public SearchBeatmapSetsRequest(string query, RulesetInfo ruleset, BeatmapSearchCategory searchCategory = BeatmapSearchCategory.Any, DirectSortCriteria sortCriteria = DirectSortCriteria.Ranked, SortDirection direction = SortDirection.Descending)
+        public SearchBeatmapSetsRequest(string query, RulesetInfo ruleset, BeatmapSearchCategory searchCategory = BeatmapSearchCategory.Any, DirectSortCriteria sortCriteria = DirectSortCriteria.Ranked, SortDirection direction = SortDirection.Descending, BeatmapSearchGenre genre = BeatmapSearchGenre.Any, BeatmapSearchLanguage language = BeatmapSearchLanguage.Any)
```

Comment:
7 params is getting a bit much. My ultrawide monitor has the horizontal scrollbar of half the size of the window when this is open. Lines should be split at least, but I'd _really_ advocate passing in an actual structure here:

```csharp
public class SearchBeatmapSetsParameters
{
    public string Query { get; set; }
    public RulesetInfo Ruleset { get; set; }
    public BeatmapSearchCategory SearchCategory { get; set; }
    public DirectSortCriteria SortCriteria { get; set; }
    public SortDirection SortDirection { get; set; }
    public BeatmapSearchGenre Genre { get; set; }
    public BeatmapSearchLanguage Language { get; set; }

    public BeatmapSetSearchParameters(string query, RulesetInfo ruleset)
    {
        Query = query;
        Ruleset = ruleset;

        SearchCategory = BeatmapSearchCategory.Any;
        SortCriteria = DirectSortCriteria.Ranked;
        SortDirection = SortDirection.Descending;
        Genre = BeatmapSearchGenre.Any;
        Language = BeatmapSearchLanguage.Any;
    }
}
```
Some of these initialisers might be a bit gratuitous as the default would be the same but probably better to be explicit than not.